### PR TITLE
Apicurio added to Avro test

### LINKED ISSUE #9478: Add Avro support to the Kafka connector
It should already work, but we need to add a test verifying that Avro can be used.

We should verify using the Confluent registry and the Apicurio registry (and custom serde). 

This feature deserves a specific section in the documentation.

Estimation: Quarkus 1.6

### PR DESCRIPTION
**Pull request by phillip-kruger**
In this PR we added Apicurio Avro Serializer and Deserializer to the Kafka connector. 
Also updated to use the latest version of Apicurio registry.

This partially fix #9478 (Documentation still outstanding)

Signed-off-by: Phillip Kruger <phillip.kruger@gmail.com>

### CODE REVIEW

**Review by cescoffier — APPROVED** [review]

**Comment by phillip-kruger** [comment]
@cescoffier native should build now. Let's see ....

File: integration-tests/kafka/pom.xml [hunk 1]
```diff
@@ -230,6 +256,7 @@
                                     <enableHttpUrlHandler>true</enableHttpUrlHandler>
                                     <graalvmHome>${graalvmHome}</graalvmHome>
                                     <enableAllSecurityServices>true</enableAllSecurityServices>
+                                    <reportErrorsAtRuntime>true</reportErrorsAtRuntime>
```
Comment (gsmet) [inline integration-tests/kafka/pom.xml:259]: I don't think we want that enabled.

**Review by gsmet — DISMISSED** [review]

File: integration-tests/kafka/pom.xml (hunk 1 above)
Comment (phillip-kruger) [reply integration-tests/kafka/pom.xml:259]: Without it the native build image creation fails die to Avro generated classes. ( Even without a native test for Avro). Any idea how I can get around it ?

Comment (phillip-kruger) [reply integration-tests/kafka/pom.xml]: As discussed with @cescoffier, we moved all Avro stuff to a separate module.

**Review by cescoffier — APPROVED** [review]

**Comment by gsmet** [comment]
Is it just me or we don't test native?

**Comment by gsmet** [comment]
Also I don't see the value in backporting it tbh. It can wait for the next release given there wasn't any fix applied.

**Comment by phillip-kruger** [comment]
Yea we don't test native because it does not work :)

**Comment by gsmet** [comment]
No opinion on this one, merged as @cescoffier approved it.

**Pull request merged**